File: lang/nodejs/test/fun/vac/algorithms/sorting/MergeSortTest.js
"use strict"

const MergeSort       = require("fun/vac/algorithms/sorting/MergeSort")
const SequenceBuilder = require("fun/vac/util/SequenceBuilder")
const Sequences       = require("fun/vac/util/Sequences")
const TestRunner      = require("fun/vac/util/TestRunner")

const testMergeSort = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packRandom(arr)

    let checksum = Sequences.parityChecksum(arr)

    MergeSort.sort(arr)

    if (Sequences.parityChecksum(arr) !== checksum) {
        return false
    }

    if (!Sequences.isSorted(arr)) {
        return false
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testMergeSort())
}

File: lang/c/test/fun/vac/algorithms/sorting/MergeSortTest.c
#include <fun/vac/algorithms/sorting/MergeSort.h>
#include <fun/vac/util/SequenceBuilder.h>
#include <fun/vac/util/Sequences.h>
#include <fun/vac/util/TestRunner.h>

bool testMergeSort(void) {
    size_t size = 32768;

    int arr[size];
    SequenceBuilder_packRandom(arr, size);

    int checksum = Sequences_parityChecksum(arr, size);

    MergeSort_sort(arr, size);

    if (Sequences_parityChecksum(arr, size) != checksum) {
        return false;
    }

    return Sequences_isSorted(arr, size);
}

int main(void) {
    TestRunner_parseTest(testMergeSort());
}
